plugins/modules/provision_workflow_manager.py

```diff
@@ -911,6 +906,21 @@ class instance for further use.
             self.log("The provisioned status of the wired device is {0}".format(status), "INFO")
 
             if status == "success":
+                if to_force_provisioning is False:
+                    self.result["changed"] = False
+                    msg = "Device '{0}' is already provisioned.".format(self.validated_config.get("management_ip_address"))
+                    self.result['msg'] = msg
+                    self.result['diff'] = self.want
+                    self.result['response'] = msg
+                    self.log(msg, "INFO")
+                    return self
+                if to_provisioning is False:
```

Comment:
add a new line before

Reply:
Your code looks good now. Can you make it to simple functions? So the inventory module can call these functions..

```
if device_type == "wired":
    self.provision_wired_device( to_provisioning, to_force_provisioning, other args if required)
    check status
elif device_type == "wireless":
    self.provision_wireless_device()
    check status
else:
   is it required?

set the status
return


def provision_wired_device(self, to_provisioning, to_force_provisioning, other args):
       check provisioned status ...

       if to_force_provisioning set to true
            reprovision_wired_device(pass args)
            return 

      if to_provisioing
           Do provision:
      else:
            assign device to site..

       do common stuffs
       return

      
 def provision_wireless_device(self, to_provisioning, to_force_provisioning, other args):



```

Reply:
updated the code